src/addon-manager.js

```diff
@@ -828,13 +971,22 @@ class AddonManager extends EventEmitter {
 
     const adapters = this.getAdaptersByPackageName(packageName);
     const adapterIds = adapters.map((a) => a.id);
+    const notifiers = this.getNotifiersByPackageName(packageName);
+    const notifierIds = notifiers.map((n) => n.id);
+
     const unloadPromises = [];
     if (adapters.length > 0) {
       for (const a of adapters) {
         console.log('Unloading', a.name);
         unloadPromises.push(a.unload());
         this.adapters.delete(a.id);
       }
+    } else if (notifiers.length > 0) {
```

Comment:
I think a single package could expose both an add-on and a notifier if it wanted to

Reply:
Yes, it could. I was considering doing that for the email adapter for backwards compat.